pkg/converter/converter_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package converter
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/test/project"
+	"github.com/istio-ecosystem/sail-operator/pkg/test/util/supportedversion"
+	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/shell"
+	. "github.com/onsi/gomega"
+)
+
+var controlPlaneNamespace = "istio-system"
+
+func TestSimpleYamlConversion(t *testing.T) {
+	g := NewWithT(t)
+
+	istioYamlText := `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`
+
+	for _, version := range supportedversion.List {
+		g.Expect(setEnvVariable("ISTIO_VERSION", version.Name)).To(Succeed(), "error adding env variable: ISTIO_VERSION")
```

Comment:
Is there any benefit to having `setEnvVariable` instead of simply calling `os.Setenv` directly?

Also, since setting the env var is not part of the test's assertions, why not simplify this to something like this:
```suggestion
		must(os.Setenv("ISTIO_VERSION", version.Name))
```

Just seems like way too much code that makes the test too verbose, taking way more time to see what the test is actually doing. It's currently hard to see what is the test setup and what are the assertions without slowly reading all the code.